pkg/binding/doc.go

```diff
@@ -13,31 +15,45 @@ most "endpoint" applications that emit or consume events.
 
 Protocol Bindings
 
-A protocol binding implements at least Message, Sender and Receiver, and usually
-Encoder.
+A protocol binding usually implements a Message, a Sender and Receiver, a StructuredEncoder and a BinaryEncoder (depending on the supported encodings of the protocol) and an Encode method.
+
+Messages and Encoding
+
+The core of this package is the Message interface. It defines the visitors for an
+encoded event in structured mode or binary mode.
+The entity who receives a protocol specific data structure representing a message (e.g. an HttpRequest) encapsulates it in a binding.Message implementation using a
+NewMessage method (e.g. http.NewMessage).
+Then the entity that wants to send the binding.Message back on the wire,
+translates it back to the protocol specific data structure (e.g. a Kafka ConsumerMessage), using
+the visitors BinaryEncoder and StructuredEncoder specific to that protocol.
+Binding implementations exposes their visitors
+through a specific Encode function (e.g. kafka.EncodeProducerMessage), in order to simplify the invocation of the
+encoding message.
+
+The encoding process can be customized in order to mutate the final result with binding.TransformerFactory. A bunch of these are provided directly by the binding/transcoder module.
+
+Usually binding.Message implementations can be encoded only one time, because the encoding process drain the message itself.
+In order to consume a message several times, the binding/buffering module provides several APIs to buffer the Message.
 
-Receiver: receives protocol messages and wraps them to implement the Message interface.
+A message can be converted to an event.Event using ToEvent method. An event.Event can be used as Message casting it to binding.EventMessage.
```

Comment:
My feeling is if you intend to use events, bindings are the wrong level to interact with this SDK.

Reply:
Agree, but this should be documented somewhere, eg to explain to a binding implementer how to implement a `Transport` starting from a binding implementation